app/src/test/java/com/imaginaryrhombus/proctimer/TimerViewModelTest.kt

```diff
@@ -1,51 +1,55 @@
 package com.imaginaryrhombus.proctimer
 
+import android.app.Application
 import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
 import com.google.firebase.FirebaseApp
 import com.imaginaryrhombus.proctimer.ui.timer.TimerViewModel
 import org.junit.Assert.assertEquals
 import org.junit.Assert.assertNotNull
+import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
-import org.robolectric.Robolectric
-import org.robolectric.RobolectricTestRunner
 
-@RunWith(RobolectricTestRunner::class)
+@RunWith(AndroidJUnit4::class)
 class TimerViewModelTest {
 
     init {
-        // これを Activity の作成前に行わないと IllegalStateException が発生する.
-        FirebaseApp.initializeApp(ApplicationProvider.getApplicationContext())
+        // JvmStatic, BeforeClass だとランタイムエラーを起こす.
+        FirebaseApp.initializeApp(ApplicationProvider.getApplicationContext<Application>())
     }
 
     /**
-     * テスト用の Activity.
+     * Activity 生成のための Rule.
      */
-    private val testActivity =
-        checkNotNull(Robolectric.setupActivity(TimerActivity::class.java)) {
-            "Activity creation for test failed."
-        }
+    @JvmField @Rule
+    val scenarioRule = ActivityScenarioRule(TimerActivity::class.java)
 
     /**
      * 秒数が正常にフォーマットされているか.
      */
     @Test
     fun testText() {
-
-        val timerViewModel = TimerViewModel(testActivity.application)
-        val method =
-            timerViewModel.javaClass.getMethod("createTimerStringFromSeconds", Float::class.java)
-
-        assertNotNull(method)
-        method.isAccessible = true
-
-        assertEquals(method.invoke(null, 30.0f), "00:30.000")
-        assertEquals(method.invoke(null, 60.0f), "01:00.000")
-        assertEquals(method.invoke(null, 75.0f), "01:15.000")
-        assertEquals(method.invoke(null, 0.0f), "00:00.000")
-        assertEquals(method.invoke(null, -10.0f), "00:00.000")
-        assertEquals(method.invoke(null, 10.110f), "00:10.110")
-        assertEquals(method.invoke(null, 60.085f), "01:00.085")
-        assertEquals(method.invoke(null, 66.666f), "01:06.666")
+        //launch(TimerActivity::class.java).use { scenario ->
+        scenarioRule.scenario.use { scenario ->
+            scenario.onActivity {
+                val timerViewModel = TimerViewModel(it.application)
+                val method =
+                    timerViewModel.javaClass.getMethod("createTimerStringFromSeconds", Float::class.java)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/fizxive/ProcTimer/blob/bfbd78506b96f982eaca31163976e3ff1f486322/app/src/test/java/com/imaginaryrhombus/proctimer/TimerViewModelTest.kt#L39"></span>Exceeded max line length (100) (cannot be auto-corrected)</td>
    </tr>
  </tbody>
</table>